packages/experience-builder-sdk/src/hooks/useDetectEditorMode.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef, useState } from 'react';
+import {
+  doesMismatchMessageSchema,
+  sendMessage,
+  tryParseMessage,
+} from '@contentful/experience-builder-core';
+import { INCOMING_EVENTS, OUTGOING_EVENTS } from '@contentful/experience-builder-core/constants';
+
+export const useDetectEditorMode = () => {
+  //Assume we are in editor mode initially if we are in iframe
+  const [isEditorMode, setIsEditorMode] = useState(inIframe());
```

Comment:
we used to have this check in the very beginning, but it led to the Hydration error when next.js app was configured to run via SSR, because on server it would have value `false` while on client it would have value `true`

Reply:
Because this is synchronous check and the hook is at the top of PreviewEditor, it should have the same behavior, but I will look into that.